plugins/guests/redhat/cap/change_host_name.rb

```diff
@@ -29,7 +29,13 @@ def self.change_host_name(machine, name)
               }
 
               # Restart network
-              service network restart
+              if (test -f /usr/bin/systemctl && systemctl -q is-active NetworkManager.service); then
+                systemctl restart NetworkManager.service
+              elif test -f /etc/init.d/network; then
+                service network restart
+              else
+                printf "Could not restart the network to set the new hostname!\n"
+              fi
```

Comment:
What about leaving the `printf` with an `exit 1` to exit with a failure?

Reply:
Yes, that should be ok.

I was trying to think about how we could show a proper Vagrant error, but that will be a larger change than what this PR needs since all the other hostname capabilities just fail if it can't set the hostname. So for now, this is ok.